server/src/main/java/org/opensearch/ingest/IngestService.java

```diff
@@ -507,6 +530,20 @@ void validatePipeline(Map<DiscoveryNode, IngestInfo> ingestInfos, PutPipelineReq
         ExceptionsHelper.rethrowAndSuppress(exceptions);
     }
 
+    public void validateProcessorCountForIngestPipeline(Pipeline pipeline) {
+        List<Processor> processors = pipeline.getCompoundProcessor().getProcessors();
```

Comment:
Should this use `flattenAllProcessors` instead as we may otherwise have compound processors which will not be accounted for in this limit?

Reply:
Agreed, there can be recursive on_failure processors which might get skipped. Updating this.